server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -270,7 +295,12 @@ private VersionValue getUnderLock(final BytesRef uid, Maps currentMaps) {
             return value;
         }
 
-        return tombstones.get(uid);
+        value = tombstones.get(uid);
+        if (value != null) {
+            return value;
+        }
+
+        return archiver.get(uid);
```

Comment:
I wonder about the order here. I *think* it works, but for `current` and `old`, we ensure to remove data upon deletes instead, i.e., we have chosen different mechanisms for consistency here. I think it at least deserves a comment.

Reply:
I've added some comment.

Reply:
I was thinking some more about this. I think the pruning of tombstones can prune tombstones before they are committed. I think that risks us from missing a delete in real time GET if following happens:

1. Delete doc with id.
2. Refresh
3. Enough time passes to let it GC (default 60s, see `IndexSettings.INDEX_GC_DELETES_SETTING`).

I think we need to keep `minDeleteTimestamp` in the archive too and use that when pruning in `LiveVersionMap.canRemoveTombstone`.

Reply:
@henningandersen I guess this is the answer to what I asked yesterday in https://github.com/elastic/elasticsearch-serverless/pull/250#discussion_r1178167184

Reply:
😀 👍